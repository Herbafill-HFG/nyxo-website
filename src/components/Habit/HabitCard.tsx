import { Link } from "gatsby"
import React, { FC } from "react"
import styled from "styled-components"
import { getIcon } from "../../Helpers/IconHelper"
import { truncate } from "../../Helpers/string-truncater"
import { Icon } from "../Icons"
import { device } from "../Primitives"

type Props = {
  period?: string | null
  title?: string | null
  slug?: string | null
  link?: boolean
  excerpt?: string | null
}

const HabitCard: FC<Props> = ({
  slug,
  title,
  period = "morning",
  excerpt = "",
  link,
}) => {
  const icon = getIcon(period)

  const content = (
    <>
      <Stripe $color={icon.secondaryColor} $accentColor={icon.color} />
      <Column>
        <Period style={{ color: `${icon.color}` }}>
          <Icon
            name={icon.name}
            height="25px"
            width="25px"
            stroke={icon.color}
          />
          {period}
        </Period>
        <Title data-testid="habit-title">{title}</Title>
        <Excerpt>{truncate(excerpt, 100, true)}</Excerpt>
      </Column>
    </>
  )
  return (
    <Card $accentColor={icon.color}>
      {link ? <Link to={`${slug}`}>{content}</Link> : content}
    </Card>
  )
}

export default HabitCard

type IconProps = {
  $color?: string
  $accentColor: string
}

const Stripe = styled.div<IconProps>`
  margin-right: 1rem;
  padding: 0.5rem;
  background-color: ${({ $color }) => $color};
  border-radius: 0.5rem;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: background-color 0.3s ease-in-out;
`

const Card = styled.div<IconProps>`
  padding: 1rem;
  margin: 1rem 0.5rem;
  display: flex;
  transition: transform 0.2s ease-in-out, box-shadow 0.2s ease-in-out;
  box-sizing: border-box;
  flex-direction: row;
  background-color: var(--secondaryBg);
  border-radius: 0.5rem;
  box-shadow: var(--shadow);

  &:hover {
    transform: translateY(-2px);

    ${Stripe} {
      background-color: ${({ $accentColor }) => $accentColor};
    }
  }

  @media ${device.desktopL} {
    max-width: calc(25% - 2 * 0.5rem);
    flex: 1 1 calc(25% - 2 * 0.5rem);
  }

  @media ${device.desktop} {
    max-width: calc(25% - 2 * 0.5rem);
    flex: 1 1 calc(25% - 2 * 0.5rem);
  }

  @media ${device.laptopL} {
    max-width: calc(33.3333% - 2 * 0.5rem);
    flex: 1 1 calc(33.3333% - 2 * 0.5rem);
  }

  @media ${device.laptop} {
    max-width: calc(33.3333% - 2 * 0.5rem);
    flex: 1 1 calc(33.3333% - 2 * 0.5rem);
  }

  @media ${device.tabletL} {
    max-width: calc(50% - 2 * 0.5rem);
    flex: 1 1 calc(50% - 2 * 0.5rem);
  }

  @media ${device.tablet} {
    max-width: calc(50% - 2 * 0.5rem);
    flex: 1 1 calc(50% - 2 * 0.5rem);
  }

  @media ${device.mobileL} {
    max-width: 100%;
    flex: 1 1 100%;
  }

  @media ${device.mobileM} {
    max-width: 100%;
    flex: 1 1 100%;
  }

  @media ${device.mobileS} {
    max-width: 100%;
    flex: 1 1 100%;
  }
`

const Period = styled.span`
  font-size: 0.8rem;
  font-family: var(--medium);
  font-style: normal;
`

const Title = styled.h4`
  margin: 0;
  font-family: var(--semibold);
  font-weight: bold;
  font-style: normal;
`
const Column = styled.div``

const Excerpt = styled.p`
  font-size: 0.9rem;
  color: var(--textSecondary);
  line-height: 22px;
`
