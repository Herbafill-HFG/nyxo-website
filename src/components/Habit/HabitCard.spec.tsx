import React from "react"
import renderer from "react-test-renderer"
import HabitCard from "./HabitCard"
import { render } from "@testing-library/react"

describe("HabitCard", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <HabitCard period="MORNING" slug="test-habit" title="Test Habit" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("displays the correct title", () => {
    const { getByTestId } = render(
      <HabitCard period="MORNING" slug="test-habit" title="Test Habit" />
    )
    expect(getByTestId("habit-title")).toHaveTextContent("Test Habit")
  })
})
