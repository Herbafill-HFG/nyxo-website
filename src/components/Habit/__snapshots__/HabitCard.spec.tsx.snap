// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCard renders correctly 1`] = `
<div
  className="sc-fzoLsD wZIUF"
>
  <div
    className="sc-fzozJi fElUoo"
  />
  <div
    className="sc-fznyAO hdDwJr"
  >
    <span
      className="sc-fzpans hbJJZj"
      style={
        Object {
          "color": "#F42D97",
        }
      }
    >
      <svg
        className="sc-AxjAm eINkHy"
        fill="none"
        height="25px"
        stroke="#F42D97"
        strokeWidth="1.5"
        version="1.1"
        viewBox="0 0 30 30"
        width="25px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M 11.728,21.457v-9.75 "
          />
          <path
            d="M 13.978,13.957l-2.25-2.25l-2.25,2.25 "
          />
          <path
            d="M 14.728,16.957h8.25 "
          />
          <path
            d="M 1.228,16.957h7.5 "
          />
          <path
            d="M 11.728,3.457v-1.5 "
          />
          <path
            d="M 18.092,6.093l1.06-1.061 "
          />
          <path
            d="M 20.728,12.457h1.5 "
          />
          <path
            d="M 5.364,6.093L4.303,5.032 "
          />
          <path
            d="M 2.728,12.457h-1.5 "
          />
          <path
            d="M 14.728,16.957 h0.93c2.504-2.171,2.774-5.96,0.603-8.464S10.301,5.72,7.797,7.89s-2.774,5.96-0.603,8.464c0.186,0.215,0.388,0.416,0.603,0.603 h0.931"
          />
        </g>
      </svg>
      MORNING
    </span>
    <h4
      className="sc-fzplWN giDusN"
      data-testid="habit-title"
    >
      Test Habit
    </h4>
    <p
      className="sc-fznKkj fyCcVX"
    >
      
    </p>
  </div>
</div>
`;
